helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -302,13 +310,14 @@ public void setInstanceDisabledReason(String disabledReason) {
    * It will be a no-op when instance is enabled.
    */
   public void setInstanceDisabledType(InstanceConstants.InstanceDisabledType disabledType) {
-    if (!getInstanceEnabled()) {
+    if (getInstanceOperation().equals(InstanceConstants.InstanceOperation.DISABLE)) {
```

Comment:
Same here. To me, just disable would be too specific as evacuate, unknown they can all have the type?

Reply:
Same as above ^